realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -504,4 +514,14 @@ private int getTotalGlobalRefCount() {
 
         return totalRefCount;
     }
+
+    /**
+     * If a Realm instance is GCed but `Realm.close()` is not called before, we still want to track the cache for
+     * debugging. Adding them to the list to keep the strong ref of the cache to prevent the cache gets GCed.
+     */
+    void leak() {
+        if (!isLeaked.getAndSet(true)) {
+            leakedCaches.add(this);
```

Comment:
aha, that is actually the case we discussed in https://github.com/realm/realm-java/pull/4551#discussion_r114300963 !!!
So shall we let user continue in this case?